### PR TITLE
Remark---massmint-multi-upload

### LINKED ISSUE #8303: follow up on massmint-multi upload
Follow Up
https://github.com/kodadot/nft-gallery/pull/8302

implement  directory pin to IPFS for Remark

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

self tested  already
regular 1 copy (plus attributes, listing for sale, support)
multiple copies (plus attributes, listing for sale, support)
massmint

all good as far as i can see

## Context

- [x] Closes #8303

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Notes

as i mentioned in Telegram, ill be away for a week
therefore i ask @kodadot/internal-dev  that you manage this PR on my behalf

1. undraft and review only after #8302 is merged
2. feel free to
 update this branch from main
commit all code suggestion and review notes

**basically you can act as if this PR is your own**

Thank You!!

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at ea1a005</samp>

This pull request adds a new feature to mint multiple tokens with a single transaction on the RMRK and Statemine chains. It refactors and improves the existing code for minting tokens, adds new functions and files for constructing and uploading metadata, and updates the tests accordingly.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at ea1a005</samp>

> _`RMRK` tokens mint_
> _Batching metadata files_
> _Autumn harvest time_

### CODE REVIEW

File: composables/transaction/mintToken/constructDirectoryMeta.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import { createMetadata, unSanitizeIpfsUrl } from '@kodadot1/minimark/utils'
```
Comment (roiLeo) [inline composables/transaction/mintToken/constructDirectoryMeta.ts:1]: why do I see this as new file while it already exist?

File: services/nftStorage.ts [hunk 2]
```diff
@@ -80,4 +80,20 @@ export const pinFileToIPFS = async (file: Blob): Promise<string> => {
...
+  const response = await nftStorageApi('/pinFile', {
+    method: 'POST',
+    body: formData,
+  }).catch((error: FetchError) => {
+    throw new Error(
+      `[NFT::STORAGE] Unable to PIN Directory for reasons ${error.data}`,
+    )
+  })
+
+  return response.value.cid
```
Comment (roiLeo) [inline services/nftStorage.ts:96]: this is same as `pinFileToIPFS`

**Review by roiLeo — APPROVED** [review]
otherwise lgtm

**Comment by prury** [comment]
RMRK2: 
minting with copies and attributes and listing ✔️ 
https://kusama.subscan.io/tx/0xf0f6566d21fe98c5da1a8293c4b95d57a3d7d2c352956457e3b71ca8ef647d4b
https://kusama.subscan.io/tx/0xf0f6566d21fe98c5da1a8293c4b95d57a3d7d2c352956457e3b71ca8ef647d4b

minting single nft right after ✔️
https://kusama.subscan.io/tx/0x7e1d0c7ff091700578de42c0d5dd9582b3de86b2f7486ef92fd2b49fc123cfe3

Massmint ✔️
https://kusama.subscan.io/tx/0x6b3326760bd8c42d0b7954b583fcb2db465a3577f88ff9cc8f80c48e0209b7c5
https://kusama.subscan.io/extrinsic/20854940-5

**Comment by yangwao** [comment]
@daiagi is iirc ooo till 10/12, can someone resolve conflicts pls 🙏

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 5.57 USD/DOT ~ 8.977 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x142ba269ab2ec764f60b252bbb4073909af7a3b0af484f4ff7953f05d7ab5e7d](https://polkadot.subscan.io/extrinsic/0x142ba269ab2ec764f60b252bbb4073909af7a3b0af484f4ff7953f05d7ab5e7d)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**